packages/squiggle-lang/src/public/SqError.ts

```diff
@@ -6,6 +6,7 @@ abstract class SqAbstractError<T extends string> {
 
   abstract toString(): string;
   abstract toStringWithDetails(): string;
+  abstract toTypeString(): string[];
```

Comment:
Allowing arbitrary number of type strings here seems excessive, or maybe I don't understand the intention; I see that you join everything except the first item with dashes in `components`, and I'm not sure if you had some longer-term vision about when more than 2 elements would be useful.

The way I'd do it is something like:
- `abstract getType(): string` for top-level error type; or, even better, change `tag` field to match that public name and use that in `components`
- separate `getSubtype(): string { return ""; }`, with `""` as a default and overridden in `SqRuntimeError`.
  - `""` instead of `undefined` for the same reason as in https://github.com/quantified-uncertainty/squiggle/pull/2542#discussion_r1396620289

(I'm using `get` prefix instead of `to` prefix in both cases, as `to` suggests "represent an entire error object as a string", while type and subtype are parts of the object)